src/librustc_platform_intrinsics/lib.rs

```diff
@@ -32,34 +33,69 @@ pub enum Type {
     Void,
     Integer(/* signed */ bool, u8, /* llvm width */ u8),
     Float(u8),
-    Pointer(Box<Type>, Option<Box<Type>>, /* const */ bool),
-    Vector(Box<Type>, Option<Box<Type>>, u8),
-    Aggregate(bool, Vec<Type>),
+    Pointer(&'static Type, Option<&'static Type>, /* const */ bool),
+    Vector(&'static Type, Option<&'static Type>, u8),
+    Aggregate(bool, &'static &'static [&'static Type]),
```

Comment:
Ah this was actually because the generated `static` values had the type `&'static [T]` and that couldn't be used by value (only by reference). I realize now though that all of the statics can have type `[T; N]` and then we can use references and coercions to get the right type, so no there's no longer double indirection.